Migrate from Slack to Discord

jbboynton opened the pull request:
Update the project so that notifications and messages are posted to
their new channel in the Land of Fires Discord server.

Discord has a very similar structure to Slack—both services allow
messages to be created via webhooks. So, the bulk of this migration
comes down to updating webhook URLs and renaming "Slack" to "Discord".

jbboynton committed:
Migrate from Slack to Discord

Update the project so that notifications and messages are posted to
their new channel in the Land of Fires Discord server.

Discord has a very similar structure to Slack—both services allow
messages to be created via webhooks. So, the bulk of this migration
comes down to updating webhook URLs and renaming "Slack" to "Discord".

jbboynton commented on src/Settings.py:
```diff
@@ -74,3 +74,43 @@ def get_username(capitalize=False):
+
+    def reset_discord_endpoints():
+        Settings.set_key(Settings.discord_prod_key, "")
+        Settings.set_key(Settings.discord_dev_key, "")
+        Settings.set_discord_endpoints()
+
+    def check_discord_endpoints():
+        settings = Settings.get_all()
+
+        if not Settings.discord_prod_key in settings or \
+           not Settings.discord_dev_key in settings:
+            return False
+
+        if settings[Settings.discord_prod_key] == "" or \
+           settings[Settings.discord_dev_key] == "":
+            return False
+
+        return True
+
+    def set_discord_endpoints():
+        if not Settings.check_discord_endpoints():
+            drive = Drive()
+
+            Settings.set_key(
+                key  = Settings.discord_prod_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_prod_key
+                )
+            )
+
+            Settings.set_key(
+                key  = Settings.discord_dev_key,
+                data = drive.get_json_key(
+                    remote_file    = f'{LOFSM_DIR_PATH}/db.json',
+                    local_filepath = f'temp/db.json',
+                    key            = Settings.discord_dev_key
+                )
+            )
```
@vindennl48 Unfortunately I couldn't figure out how to put this behavior in `Discord.py` like we discussed. I don't see a way to avoid the circular dependency, and I don't think I fully understood how to implement this like you were suggesting. Could you point me in the right direction?

vindennl48 replied:
For future reference, we found a bug in Settings that it was referencing Settings.get_slack_endpoints and that function doesn't exist.  It was moved to Slack.get_slack_endpoints, and that function was never called to cause an issue.  We will keep it in Settings class for now.

jbboynton reviewed (commented):
@vindennl48 Here's my second attempt at this change. This time I tried to change as little as possible—based on what we chatted about earlier, I'm hoping this PR is closer to the mark.

vindennl48 reviewed (approved):
Looks good.  I will test and merge shortly.

vindennl48 committed:
Added migration file and changed settings class

Created the migration 1_4 file to create the keys for the discord
endpoints inside the .settings file.  This migration also removes the
old slack endpoints as well as the 'user' key that was deprecated a bit
ago.

Changed the discord set endpoints function in Settings class to allow
for manual entry.  The automatic way was not finished and needed quite a
bit more work to get running so we are defaulting back to this.
